doc/user/content/sql/types/list.md

```diff
@@ -0,0 +1,490 @@
+`list` of `int list`s, i.e. a 2D list.
+
+### Construction
+
+You can construct lists using the `LIST` expression:
+
+```sql
+SELECT LIST[1, 2, 3];
+```
+```nofmt
+  list
+---------
+ {1,2,3}
+```
+
+You can nest `LIST` constructors to create multidimensional lists:
+
+```sql
+SELECT LIST[LIST['a', 'b'], LIST['c']];
+```
+```nofmt
+    list
+-------------
+ {{a,b},{c}}
+```
+
+You can also elide the `LIST` keyword from the interior list expressions:
+
+```sql
+SELECT LIST[['a', 'b'], ['c']];
+```
+```nofmt
+    list
+-------------
+ {{a,b},{c}}
+```
+
+Multidimensional lists can be "ragged", i.e. the length of each nested list
+can differ from others. This differs from `array`, which does not
+support ragged multideimensional arrays, i.e. the length of each nested array
+must be the same.
+
+Note that you can also construct lists using the available [`text`
+cast](#text-to-list-casts).
+
+### Accessing lists
+
+#### Elements (subscripting)
+
+Accessing elements of `list`s (known as subscripting) uses brackets (`[]`)
+and 1-index element positions:
+
+```sql
+SELECT LIST[['a', 'b'], ['c']][1];
+```
+```nofmt
+ ?column?
+----------
+ {a,b}
+```
+
+Subscripts can be chained together to descend the list's dimensions:
+
+```sql
+SELECT LIST[['a', 'b'], ['c']][1][2];
+```
+```nofmt
+ ?column?
+----------
+ b
+```
+
+If the index is invalid (either less than 1, or greater the maximum index),
+lists return _null_.
+
+```sql
+SELECT LIST[['a', 'b'], ['c']][1][5] AS exceed_index;
+```
+```nofmt
+ exceed_index
+--------------
+
+```
+
+Lists have types based on their dimension (unlike arrays), and error if you
+attmept to subscript a non-list element (i.e. the depth of subscripting exceeds
+the list'd maximum dimension):
+
+```sql
+SELECT LIST[['a', 'b'], ['c']][1][2][3];
+```
+```nofmt
+ERROR:  cannot subscript type string
+```
+
+#### Ranges/slices (subscripting)
+
+Slicing `lists` (confusingly _also_ known as subscripting) uses the format
+`[first index : last index]`, using 1-indexed positions:
+
+```sql
+SELECT LIST[1,2,3,4,5][2:4] AS two_to_four;
+```
+```nofmt
+  slice
+---------
+ {2,3,4}
+```
+
+You can omit the first index to use the first value in the `list`, and omit the
+last index to use all elements remaining in the `list`.
+
+```sql
+SELECT LIST[1,2,3,4,5][:3] AS one_to_three;
+```
+```nofmt
+ one_to_three
+--------------
+ {1,2,3}
+```
+
+```sql
+SELECT LIST[1,2,3,4,5][3:] AS three_to_five;
+```
+```nofmt
+ three_to_five
+---------------
+ {3,4,5}
+```
+
+If the first index exceeds the list's maximum index, the operation returns _null_:
+
+```sql
+SELECT LIST[1,2,3,4,5][10:] AS exceed_index;
+```
+```nofmt
+ exceed_index
+--------------
+
+```
+
+If the last index exceeds the list's maximum index, the operation returns all
+remaining elements up to its final element.
+
+```sql
+SELECT LIST[1,2,3,4,5][2:10] AS two_to_end;
+```
+```nofmt
+ two_to_end
+------------
+ {3,4,5}
+```
```

Comment:
What do you think about calling the first operation "indexing", the second operation "slicing", and using "subscripting" as the umbrella term to refer to either indexing or slicing?